### PR TITLE
Create budget model and budget model specs, add error messages to con…

### PR DESCRIPTION
**Pull request by apalski**
…fig/locales/en.yml, run rubocop and fix violations
PR request for budget model, validations, budget model specs.
Add error messages for translations to en.yml.
Ran Rubocop and fix violations.

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -17,3 +17,12 @@ en:
...
     application: Budgetapp
+
+  activerecord:
+    errors:
+      models:
+        budget:
+          attributes:
+            name:
+              blank: "must be entered"
+              taken: "already taken"
```
Comment (elle) [inline config/locales/en.yml]: You can use a more generic format for the yams structure, to remove the duplication in error messages.
Check the module on translations in the curriculum.

File: db/migrate/20180120013036_create_budgets.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateBudgets < ActiveRecord::Migration[5.1]
+  def change
+    create_table :budgets do |t|
+      t.string :name, null: false
+      t.string :budget_type, default: "basic"
```
Comment (elle) [inline db/migrate/20180120013036_create_budgets.rb]: I would consider using Enums for `budget_type`, which means line 5 would have to change to:
`t.integer, :budget_type, default: 0`

Check the curriculum for enums

File: spec/models/budget_spec.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
+require "rails_helper"
+require "./app/models/budget"
```
Comment (elle) [inline spec/models/budget_spec.rb]: Not needed because of `rails_helper`

File: spec/models/budget_spec.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+describe Budget do
+  it "can be created" do
+    budget = Budget.create(name: "holiday")
+    budget.save!
+
+    expect(budget).to be_a Budget
+  end
```
Comment (elle) [inline spec/models/budget_spec.rb]: You are testing core Rails functionality here. I don't think this spec is needed.

**Comment by elle** [comment]
One more thing: PR description can be improved :)

File: config/locales/en.yml (hunk 1 above)
Comment (apalski) [reply config/locales/en.yml]: Thanks Elle, will look at this and the factory above - great suggestions!
Ignore in the next PRs as I will take this on for all models!

File: db/migrate/20180120013036_create_budgets.rb (hunk 2 above)
Comment (apalski) [reply db/migrate/20180120013036_create_budgets.rb]: Mmm, I like this - thanks Elle - not sure yet though if I want this to be customisable by the user, ie. let the name it whatever they want in line with why they are creating it.

Comment (elle) [reply db/migrate/20180120013036_create_budgets.rb]: If you wish to have set types of budgets, enum makes sense.

**Comment by apalski** [comment]
Hi @elle 
I think this PR is ready for review. Thanks again for your help!

File: db/migrate/20180120013036_create_budgets.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+    end
```
Comment (elle) [inline db/migrate/20180120013036_create_budgets.rb]: Missing timestamps

File: spec/factories.rb [hunk 6]
```diff
@@ -1,2 +1,5 @@
 FactoryBot.define do
+  factory :budget do
+    sequence(:name) { |n| "name#{n}" }
```
Comment (elle) [inline spec/factories.rb]: A suggestion to change to "budget#{n}" which will help with debugging what objects you work with in the specs

File: spec/models/budget_spec.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  context "budget_type" do
+    it "has a default budget type" do
+      budget = Budget.new
+
+      expect(budget.budget_type).to eq("basic")
+    end
+
+    it "defines a custom budget type" do
+      budget = build(:budget, budget_type: "Ireland")
```
Comment (elle) [inline spec/models/budget_spec.rb]: 👍

File: config/routes.rb [hunk 8]
```diff
@@ -1,2 +1,6 @@
 Rails.application.routes.draw do
+  scope "/:locale", locale: /en/ do
+    resources :users, only: [:new]
+    resources :budgets, only: [:new]
```
Comment (elle) [inline config/routes.rb]: These are not needed yet, cause there aren't any controllers and views in this PR.
Also, is the local scope still needed?

**Pull request closed without merging**